content/2017.03.building-projects-as-a-way-to-learn.rst

```diff
@@ -0,0 +1,200 @@
+already.
+
+I've learned a lot, and with the storage team we have been part of a
+number of different projects:
+
+- Firefox Hello: Building a WebRTC signaling service.
+- MSISDN auth: Building a password less BrowserID authentication
+  service using a phone number.
+- Syncto: Browsing Firefox Sync using the Kinto protocol.
+- Kinto: Building synchronization services with backup and sharing capability in mind.
+- RemoteServiceWorker: Building a Firefox Service Worker in the cloud
+- WebPush channels: Building a PubSub on top of WebPush
+
+While most of these projects have been built, deployed in production
+and eventually decommissined, some of them are still actively use by
+Firefox users everyday.
+
+
+How do you keep motivated building things that are thrown away?
+===============================================================
+
+Let's be honest, everything we build: cars, houses, and even softwares
+are meant to be eventually thrown away. That's why recycling is so
+important.
+
+The question is why do we keep building them nevertheless.
+
+Our daily job is to help Mozilla make the web a safe place for users
+and prevent big companies to turn them into advertisement products
+without their consent.
+
+One way of doing so is to build software that empower people to use
+internet while being in charge of their own data.
+
+When users give their data to big companies, they are given them
+leverage. In the meantime these companies are offering a service that
+solve people problems, and they get addicted to it and come back for
+more.
+
+We alone cannot compete with companies that are spending millions of
+dollars building great products to capture users.
+
+However, we can build software that, in one hand, is useful for the
+Mozilla mission and makes us reactive to fix Mozilla issues while
+building Firefox, and in the other hand enpower developers to build
+great software where users choose where they want to store their data.
+
+What matters rather than building software is the vision that this
+software carries.
+
+Writting lines of code is cheap and fast. You can build a prototype in
+two weeks and iterate until you are happy with the result.
+
+Then it might be use by some people and it might get some traction for
+few months or few years and it will be eventually forgotten in a git
+repository somewhere forever.
+
+Writting software to solve problems is not enough, to keep motivated
+on the long term.
+
+Our understanding is that you need a greater goal than just writting
+code.
+
+For us the answer to that question is: what can we learn while
+building software that can be useful for us and for our community to
+help us go faster and build more robust software later.
+
+For Mozilla the answer to that question is: How do I improve Firefox
+so that people want to use it as their main browser?
+
+Both answer are goals that we set to ourselves and we are using
+software development to reach them.
```

Comment:
These paragraphs here use "answer" and "question" in quite confusing ways, and relate them to "goals". I think this could be cleaned up a bit.